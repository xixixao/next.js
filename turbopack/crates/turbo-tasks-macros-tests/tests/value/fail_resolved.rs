#![feature(arbitrary_self_types)]

use turbo_tasks::Vc;

#[turbo_tasks::value(resolved)]
struct MyValue {
    value: Vc<i32>,
}

fn main() {
    let v = MyValue { value: Vc::cell(0) };
    let _ = v.value;
}
