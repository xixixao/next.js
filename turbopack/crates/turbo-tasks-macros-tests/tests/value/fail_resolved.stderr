error[E0277]: the trait bound `Vc<i32>: ResolvedValue` is not satisfied
 --> tests/value/fail_resolved.rs:7:12
  |
7 |     value: Vc<i32>,
  |            ^^^^^^^ the trait `ResolvedValue` is not implemented for `Vc<i32>`
  |
  = help: the following other types implement trait `ResolvedValue`:
            &T
            &mut T
            ()
            (A, Z, Y, X, W, V, U, T)
            (B, A, Z, Y, X, W, V, U, T)
            (C, B, A, Z, Y, X, W, V, U, T)
            (D, C, B, A, Z, Y, X, W, V, U, T)
            (E, D, C, B, A, Z, Y, X, W, V, U, T)
          and $N others
note: required by a bound in `DeriveResolvedValueAssertion::assert_impl_resolved_value`
 --> tests/value/fail_resolved.rs:5:22
  |
5 | #[turbo_tasks::value(resolved)]
  |                      ^^^^^^^^ required by this bound in `DeriveResolvedValueAssertion::assert_impl_resolved_value`
  = note: this error originates in the derive macro `turbo_tasks::ResolvedValue` (in Nightly builds, run with -Z macro-backtrace for more info)
